File: tests/fixtures/vulnerable/no_auth_routes.ts
// Vulnerable: Express routes without auth middleware
import express from "express";

const app = express();

app.get("/api/users/:userId", (req, res) => {
  // No auth middleware
  res.json(getUserData(req.params.userId));
});

app.post("/api/admin/settings", (req, res) => {
  // No auth on admin route
  res.json({ status: "updated" });
});

// Public route (should NOT flag)
app.get("/health", (req, res) => {
  res.json({ status: "ok" });
});

File: tests/fixtures/vulnerable/no_auth_routes.py
"""Vulnerable: Flask routes without auth decorators."""
from flask import Flask, jsonify

app = Flask(__name__)


@app.route("/api/users/<user_id>")
def get_user(user_id):
    """Bad: no auth on sensitive route."""
    return jsonify(get_user_data(user_id))


@app.route("/api/admin/settings", methods=["POST"])
def update_settings():
    """Bad: admin route without auth."""
    return jsonify({"status": "updated"})


@app.route("/health")
def health():
    """OK: public health endpoint."""
    return {"status": "ok"}
